rosbag2/src/rosbag2/writer.cpp

```diff
@@ -94,6 +137,18 @@ void Writer::write(std::shared_ptr<SerializedBagMessage> message)
     throw std::runtime_error("Bag is not open. Call open() before writing.");
   }
 
+  // Update the message count for the Topic.
+  ++topics_names_to_info_.at(message->topic_name).message_count;
+
+  const auto message_timestamp = std::chrono::time_point<std::chrono::high_resolution_clock>(
+    std::chrono::nanoseconds(message->time_stamp));
+
+  metadata_.starting_time = std::min(metadata_.starting_time, message_timestamp);
+
+  const auto duration = message_timestamp - metadata_.starting_time;
+
```

Comment:
nitpick: consider removing the line

Reply:
I think defining the local variables `message_timestamp` and `duration` improves the readability of this code block.
The alternative is something like:

```c++
metadata_.starting_time = std::min(
  metadata_.starting_time,
  std::chrono::time_point<std::chrono::high_resolution_clock>(
    std::chrono::nanoseconds(message->time_stamp)));

metadata_.duration = std::max(
  metadata_.duration,
  std::chrono::time_point<std::chrono::high_resolution_clock>(
    std::chrono::nanoseconds(message->time_stamp) - metadata_.starting_time);
```

Reply:
comment was meant for the empty lines.